### PR TITLE
Indexes

### PR DESCRIPTION
**Pull request by J-avery32**
This implements indexing. I recommend merging #94 first and running its upgrade script against the database and then merging this branch. In fact slack me when you are going to try and merge this branch because there are probably some oddities in Mongo's data right now that we will need to fix before building some of the unique indexes.

### CODE REVIEW

**Comment by J-avery32** [comment]
fixes: #85

File: splash/api/indexes.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+from pymongo import IndexModel
+from pymongo import ASCENDING, TEXT
+
+
+def create_indexes_pages_old(collection):
```
Comment (dylanmcreynolds) [inline splash/api/indexes.py:5]: can we change to `create_indexes_historic_pages`

File: splash/service/base.py [hunk 2]
```diff
@@ -36,9 +37,10 @@ def delete(self, current_user: User, uid):
 
 
 class MongoService:
-    def __init__(self, db, collection_name):
+    def __init__(self, db, collection_name, create_indexes):
```
Comment (dylanmcreynolds) [inline splash/service/base.py:40]: consider creating `def _create_indexes()` and calling it in subclasses

**Review by dylanmcreynolds — COMMENTED** [review]
Create indexes for the places where we are creating default sorts

**Pull request merged**